language-server/vscoqtop/lspManager.ml

```diff
@@ -211,23 +213,27 @@ let update_view uri st =
     publish_diagnostics uri st;
   )
 
+let replace_state path st visible = Hashtbl.replace states path { st; visible }
+
 let run_documents () =
-  let interpret_doc_in_bg path (st : Dm.DocumentManager.state) events =
-    let st = Dm.DocumentManager.clear_observe_id st in
-    let (st, events') = Dm.DocumentManager.interpret_in_background st in
-    let uri = DocumentUri.of_path path in
-    Hashtbl.replace states path st;
-    update_view uri st;
-    let events' = inject_dm_events (uri, events') in
-    events@events'
+  let interpret_doc_in_bg path { st : Dm.DocumentManager.state ; visible } events =
+    if not visible then events
```

Comment:
This if makes me thing the current code does stuff on closed tabs, no?

Reply:
Nope, it is OK to do the things on invisible docs. For example if a settings change.